content/blog/living-off-the-ipad-as-an-engineer/index.md

```diff
@@ -0,0 +1,110 @@
+
+*How the heck do we write and run our beloved code, Pierre? 💁*
+
+As it’s common these days, the trick is in the cloud. While it comes with its drawbacks such as with spotty connections, offloading the work to a remote, safe, always-accessible machine has some nice things going for it. A hosted instance can be fired in minutes these days. To stay somewhat minimalist, I chose to keep an old laptop plugged in a closet, but there are many options out there.
+
+There’s essentially two solutions that work for me at the moment: [Visual Studio Code](https://code.visualstudio.com/) in the browser, which can be achieved in various ways; or a native SSH client app such as [Termius](https://apps.apple.com/us/app/termius-ssh-client/id549039908) or [Blink Shell](https://blink.sh/). The former is a cross-platform cloud SSH/Mosh service that has really good iPad support, its main downside being the lack of a full external monitor integration, as mentioned earlier. Blink Shell features it, and it works quite well. Picture that clean tmux/vim session on a 4K monitor, at the cost of losing mouse support, though.
+
+![A mobile web dev session in the native app Termius, with the side-kick browser](termius.png)
+
+For a proper IDE, there are ways to run the undisputed leader of the last years—VS Code — in Safari. I personally only fire it for refactoring and large Git diffs, I’m happy with terminal-only work otherwise. [code-server](http://code-server.dev/) is a project that does exactly that, and at the time of writing comes with a Beta parameter --link [that does all the hard work of securing the connection](https://github.com/cdr/code-server#cloud-program-%EF%B8%8F). One line to install on various hosts, and one port-forwarding command to start the remote access. For more details on the setup with more technical bits, you can check out [my Notion project page](https://www.notion.so/iPad-Challenge-ba216d5956194453a0dd4d56f62d888c).
+
+The big downside here is the [lack of scrolling support](https://github.com/microsoft/vscode/issues/106232). Lucky for me, I’m navigating keyboard-only with the help of the [Vim extension](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim) ([Colemak version](https://marketplace.visualstudio.com/items?itemName=ollyhayes.colmak-vim)), but it’s a real barrier to entry. It’s related to a bug in the web engine. Nonetheless, there are a few hosted solutions with the same problem, such as [Stackblitz](https://stackblitz.com/) and [GitHub Codespaces](https://github.com/features/codespaces)—which both get honourable mentions yet aren’t open source—so I’m confident that issue might get solved soon.
+
+In the example below, it runs in Safari for iPad. Still, it would be a great example where a Shiftscreen pane would help having both VS Code on one side and whatever docs on the other for a true multitasking experience on an external display.
```

Comment:
In the example below shows VS Code running in Safari for iPad. Still, it would be a great if a Shiftscreen pane could have both VS Code on one side and whatever docs on the other for a true multitasking experience on an external display.

Reply:
I believe the intent was to replace “In the example” with “The example”. Also the meaning might have changed after “Still,”. The idea is that Shiftscreen in the case of VS Code would allow a real multitasking experience on the external monitor.
```suggestion
The example below shows VS Code running in Safari for iPad. It would be a great use case for Shiftscreen, which could have both VS Code and whatever docs on the side for a true multitasking experience on an external display.
```